### PR TITLE
Fix duplicate SoundFont listing in mixer /listmidi

### LINKED ISSUE #3046: Confusing `MIXER /listmidi` output if symlink used for soundfont
### Are you using the latest Dosbox-Staging Version?

- [X] I have checked releases and am using the latest release.

### Different version than latest?

0.80.1, current GIT main (6e534ac) 

### What Operating System are you using?

Linux x86_64

### If Other OS, please describe

Gentoo

### Relevant hardware info

_No response_

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

I use a `default.sf2` symlink for the soundfont I want to use in all the emulators and MIDI players - this way I can change the soundfont systemwide without configuring each and every application separately. Unfortunately, this causes display glitches in `MIXER /listmidi` output, it shows two files as set - in 0.80.1:

screenshot-0 80 1

and in the code from PR https://github.com/dosbox-staging/dosbox-staging/pull/3011:

screenshot-devel

IMHO it shouldn't duplicate the soundfont file in the command output, and just one (`default.sf2`) should be marked as being set.

### Steps to reproduce the behaviour.

1. Put some soundfonts into `/usr/share/sounds/sf2` directory.
2. In this directory create a symlink to a soundfont, for example:

```bash
cd /usr/share/sounds/sf2
ln -s ./FluidR3_GM_GS.sf2 default.sf2
```

3. In DOSBox configuration file enable FluidSynth as a MIDI device, set the symlink as a soundfont.
4. Start DOSBox, type `MIXER /listmidi`

### Download URL of affected game or software

_No response_

### Your configuration

```ini
[midi]
mididevice = fluidsynth

[fluidsynth]
soundfont = /usr/share/sounds/sf2/default.sf2
```

### Provide a Log

_No response_

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

### PR DESCRIPTION
**Pull request by weirddan455**
# Description
The `mixer /listmidi` output would sometimes show a duplicate entry.  This happens when the user provides an absolute path that is inside one of the default soundfont directories (ex. `/usr/share/soundfonts`).  In that case, it gets printed twice.  Once from parsing the config file and once from iterating over the default directories to search.

I added a check to prevent this and also made a few small cleanups along the way.  I also now find the currently configured SoundFont in the same way that it is found when opening FluidSynth.  The active SoundFont gets printed first and then compared when iterating over directories so it does not get printed a second time.

## Related issues
Fixes #3046

# Manual testing
Followed repo steps in #3046 and confirmed the duplicate entry no longer happens.  It was not specific to symlinks.  The bug also happens if you provide an absolute path to a regular file inside the default soundfont directory.

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [x] commented on the particularly hard-to-understand areas of my code.
- [x] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: src/midi/midi_fluidsynth.cpp [hunk 1]
```diff
@@ -856,15 +856,23 @@ std::string format_sf2_line(size_t width, const std_fs::path& sf2_path)
 
 MIDI_RC MidiHandlerFluidsynth::ListAll(Program* caller)
 {
+	// Find SoundFont from user config. FluidSynth may not be Open so it
```
Comment (johnnovak) [inline src/midi/midi_fluidsynth.cpp:859]: Minor: lowercase "open"? Or "active"?

File: src/midi/midi_fluidsynth.cpp [hunk 2]
```diff
@@ -856,15 +856,23 @@ std::string format_sf2_line(size_t width, const std_fs::path& sf2_path)
...
 	auto* section = static_cast<Section_prop*>(control->GetSection("fluidsynth"));
 	const auto sf_spec = parse_soundfont_pref(section->Get_string("soundfont"));
-	const auto sf_name      = std::get<std::string>(sf_spec);
+	const std_fs::path found_soundfont = find_sf_file(
+	        std::get<std::string>(sf_spec));
+
 	const size_t term_width = INT10_GetTextColumns();
 
-	auto write_line = [caller](bool do_highlight, const std::string& line) {
+	auto write_line = [caller, term_width, this](const std_fs::path& sf2_path) {
```
Comment (johnnovak) [inline src/midi/midi_fluidsynth.cpp:868]: Just use `[&]` because you've listed `this` anyway?

File: src/midi/midi_fluidsynth.cpp (hunk 1 above)
Comment (weirddan455) [reply src/midi/midi_fluidsynth.cpp]: Fixed

**Review by johnnovak — APPROVED** [review]
Thanks for addressing this @weirddan455

File: src/midi/midi_fluidsynth.cpp (hunk 2 above)
Comment (weirddan455) [reply src/midi/midi_fluidsynth.cpp]: Sure.  I always forget the syntax for lambdas.  Changed as you suggested.

**Pull request merged**